pkg/codegen/python/gen.go

```diff
@@ -3165,7 +3165,7 @@ func ensureValidPulumiVersion(requires map[string]string) (map[string]string, er
 	// This is just legacy functionality; there's no obvious reason this should be the case.
 	if len(requires) == 0 {
 		result := map[string]string{
-			"pulumi": "",
+			"pulumi": ">=3.0.0,<4.0.0",
```

Comment:
Consider moving this to a const like we do in Node.js codegen (although, probably doesn't need to be exported) and using that here and on line 3175 below:

https://github.com/pulumi/pulumi/blob/bebe5a9b903ecf964629c4041ccb1389955eb530/pkg/codegen/nodejs/gen.go#L49